packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingMethod.gql.js

```diff
@@ -36,11 +38,20 @@ export const SET_SHIPPING_METHOD = gql`
             cart {
                 id
                 ...SelectedShippingMethodFragment
+
+                # Include fragments that update the price summary.
+                ...ShippingSummaryFragment
+                prices {
+                    ...GrandTotalFragment
+                }
```

Comment:
On the cart page we approach this a bit differently. In [that mutation](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CartPage/PriceAdjustments/ShippingMethods/shippingRadios.js#L71) we re-fetch the entire cart data. I prefer this more granular approach, and Apollo's own docs even suggest that you should be specific about the data you are updating.

With that said, I think it is a good idea to use the entire `priceSummaryFragment` here rather than specific pieces of it.

Reply:
Good call. Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/3d0cadee0db37574495b52671a12be158ab36fea.